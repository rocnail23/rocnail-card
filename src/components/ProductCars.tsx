import { useProduct } from "../hooks/useProduct"
import React,{ createContext, ReactElement,CSSProperties } from 'react';
import styles from "../styles/styles.module.css"
import { ContextValues, InitialValues, OnChangeArgs, Product, ProductCardHandlers } from '../interfaces/interfaces';




export const CarsContex = createContext({} as ContextValues)
const {Provider} = CarsContex

export interface ProductProps  {
  product : Product,
  /* children: ReactElement | ReactElement[], */
  children: (event: ProductCardHandlers) => ReactElement
  className?: string,
  style?: CSSProperties,
  onChange?: (args: OnChangeArgs) => void,
  value?: number, 
  initialValues ?: InitialValues
}

export const ProductCars = ({product,children,className,style,onChange,value,initialValues}: ProductProps) => {

  const {increaseBy,state,reset} = useProduct({onChange,product,value,initialValues})
  
  


  return (
    <Provider value={{
        product,
        increaseBy,
        state,
        maxCount: initialValues?.maxCount
    }}>
    <div className={`${styles.productCard} ${className}`}
    style={style}>

      {children({
        count:state,increaseBy,
        product,
        maxCount:initialValues?.maxCount,
        isMaxCountReached: state == initialValues?.maxCount,
        reset

      })}
    </div>
    </Provider>
  )
}


