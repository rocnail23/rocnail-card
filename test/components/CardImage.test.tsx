import renderer from "react-test-renderer"
import React from "react"
import ProductCars, { ProductImage } from "../../src/components"

describe("pruebas en productImages", () => {


    test("mostrar url en props" ,() => {


        const wraper = renderer.create(
            <ProductImage image="https/cafe.com"/>
        )


        expect(wraper.toJSON()).toMatchSnapshot()

    })

    test("mostrar url en pasado por product" ,() => {


        const wraper = renderer.create(
            <ProductCars product={{
                id:1,
                title:"cafe",
                image:"https/thisiscoffe",

            }}>
            {() => (
                <ProductImage/>
            )}
            </ProductCars>
        )


        expect(wraper.toJSON()).toMatchSnapshot()

    })


})