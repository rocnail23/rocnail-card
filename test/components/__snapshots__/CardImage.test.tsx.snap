// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en productImages mostrar url en pasado por product 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="coffe mug"
    className="productImg undefined"
    src="https/thisiscoffe"
  />
</div>
`;

exports[`pruebas en productImages mostrar url en props 1`] = `
<img
  alt="coffe mug"
  className="productImg undefined"
  src="https/cafe.com"
/>
`;
